docker/test/integration/steps/steps.py

```diff
@@ -546,3 +554,14 @@ def step_impl(context, query, number_of_rows, timeout_seconds):
 @then("the Minifi logs contain the following message: \"{log_message}\" in less than {duration}")
 def step_impl(context, log_message, duration):
     context.test.check_minifi_log_contents(log_message, timeparse(duration))
+
+# MQTT
+@then("the MQTT broker has 1 log line matching \"{log_pattern}\"")
```

Comment:
this sounds like the test will only pass if there is exactly 1 matching log line, but in fact we check for at least 1 matching log line
```suggestion
@then("the MQTT broker has a log line matching \"{log_pattern}\"")
```